src/mlpack/bindings/go/mlpack/cli_util.go

```diff
@@ -0,0 +1,145 @@
+package mlpack
+
+/*
+#cgo CFLAGS: -I. -I/capi -g -Wall
+#cgo LDFLAGS: -L/usr/local/lib -lm -L/usr/local/lib/ -lmlpack -lgo_util
```

Comment:
Hi @rcurtin  I want some help in this part. Hence when i was testing Go binding I have to set these `LDFLAGS` manually and also have to copy all `.so` file at one place for testing purpose. Copying this files is not a big issue but setting this manually creates problem while testing with the help of  `.travis.yml`.

And Sorry for the delay I was travelling from last couple of weeks. 
Thank You :)

Reply:
No worries about the delay. :+1:

I'm not sure I understand the problem fully.  I thought that the instructions from Yasmine in #1492 could be used to use the Go bindings without copying all the .so files, and if any of the Go bindings depended on libraries in non-standard places, the `LDFLAGS` in the .go files would point them at the right place.  Can you explain more of the issue?  If you can give me enough instructions to reproduce it I can try it also and probably give better feedback.

Reply:
When I ran the test with this `LDFLAGS` I go this error:
```
# mlpack/build/src/mlpack/bindings/go/mlpack
/usr/bin/ld: warning: libmlpack.so.3, needed by src/mlpack/bindings/go/mlpack/libgo_util.so, may conflict with libmlpack.so.2
?   	mlpack/build/src/mlpack/bindings/go/mlpack	[no test files]
# mlpack/build/src/mlpack/bindings/go/tests.test
/usr/bin/ld: warning: libmlpack.so.3, needed by /home/asus/go/src/mlpack/build/src/mlpack/bindings/go/mlpack/libgo_util.so, may conflict with libmlpack.so.2
signal: segmentation fault (core dumped)
FAIL	mlpack/build/src/mlpack/bindings/go/tests	0.138s
```  
But When I change the `-LFLAGS` to `-L/home/asus/go/src/mlpack/build/lib` and copy all `.so` to `/home/asus/go/src/mlpack/build/lib` it worked.

Here We can also copy all `libmlpack.so` to `/home/asus/go/src/mlpack/build/src/mlpack/bindings/go/mlpack/` and Set -LDFLAGS  as `-L.` and aslo set `LD_LIBRARY_PATH=/home/asus/go/src/mlpack/build/src/mlpack/bindings/go/mlpack/` than it work.

Reply:
Oh, I see.  Now I am remembering that when I got this running, the key was that we installed mlpack to `/usr/local/lib/`, and then it ran correctly.  However I can see from your output that you have mlpack 2 installed in `/usr/local/lib` or `/usr/lib`, which definitely is going to cause a crash.  Changing `LDFLAGS` to point at your build directory's `lib/` directory will work, but I'm a bit confused when you say you had to copy all `.so` files there too---can you tell me which .so files you mean there?

At the end of the day it's okay if some special things have to be done to run the Go bindings in the build environment, but mostly we just need to make sure that when the bindings are deployed in a Go environment, that we can have all the dynamic library path stuff handled automatically.

Reply:
all `.so` that build while making go binding using cmake eg: `libgo_util.so`, 'libmlpack_go_test_go_binding.so', 'libmlpack_go_pca.so', etc.

Reply:
Ok, I see.  Yeah, those will need to be somewhere where they can be found by the dynamic runtime linker.  Perhaps we should have CMake install them by adding an `install()` line to the CMake configuration of `src/mlpack/bindings/go/CMakeLists.txt` at line 251 or thereabouts?  Here is the `install()` command for the `mlpack` target, which you could adapt:

```
install(TARGETS mlpack
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

I am not sure if that would be the best long-term solution, but I think it should be enough to at least get something working here.  Also be sure to uninstall mlpack 2 from your system; it's likely to cause hard-to-find problems. :)

Reply:
Sure I will use and let you know.
Thank You :)

Reply:
I have added this part to the cmakelists.txt.